sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/io/AvroIO.java

```diff
@@ -326,6 +327,14 @@
       }
 
       @Override
+      public void populateDisplayData(DisplayData.Builder builder) {
+        builder.add("schema", type);
+        if (filepattern != null) {
+          builder.add("filePattern", filepattern);
+        }
```

Comment:
We allow annotating other "components" with the `HasDisplayData` interface, and then you can include within a transform via:

    builder.include(mySource);

This is particularly useful if a transform is constructed with some sub-components which are a black box. If the transform knows the relevant metadata that the component is using, it may be easier for the transform to specify. Either implementation is fine.